src/yunohost/tools.py

```diff
@@ -565,3 +565,26 @@ def tools_diagnosis(auth, private=False):
         diagnosis['private']['domains'] = domain_list(auth)['domains']
 
     return diagnosis
+
+
+def tools_public_key():
+    key_type = 'ed25519'
+    keys_folder = '/etc/yunohost/keys'
+    key_path = os.path.join(keys_folder, '/id_{}_ynh'.format(key_type))
+    pub = os.path.exists(key_path + '.pub')
+    private = os.path.exists(key_path)
+
+    if pub and not private:
+        os.remove(key_path + '.pub')
+        pub = False
+    elif not pub and private:
+        os.remove(key_path)
+        private = False
+
+    if not pub and not private:
+        if not os.path.exists(keys_folder):
+            os.mkdir(keys_folder)
+        os.system('ssh-keygen -q -t {0} -N "" -C "Generated by Yunohost" -f {1}'.format(key_type, key_path))
```

Comment:
We should probably uses this https://docs.python.org/2/library/subprocess.html#subprocess.check_call to ensure that the call was correctly made.